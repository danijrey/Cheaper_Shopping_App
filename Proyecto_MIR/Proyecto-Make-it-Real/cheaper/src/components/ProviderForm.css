.App {
	width: 100vmin;
	height: 100vmin;
	text-align: center;
}

.FormContainer {
	width: 100vw;
	height: 100vh;
	background: linear-gradient(rgba(0, 0, 0, 0.5), rgba(0, 0, 0, 0.1)),
		url("../assets/Carros.jpg") no-repeat center top;
	background-size: cover;
	color: white;
	display: flex;
	flex-direction: column;
	align-items: center;
}

.NavBar-Logo-ProvidersCreate {
	width: 300px;
	margin: 15px;
}

.TitleProvider {
	font-size: 24px;
	margin-bottom: 5px;
	margin-top: 5px;
	text-align: center;
}

.PaginaNew-Provider {
	width: 270px;
	height: 290px;
	margin: 20px;
	background-color: rgba(48, 48, 48, 0.8);
	border-radius: 10px;
}

.InputsNew-Provider {
	display: flex;
	justify-content: center;
}

.ProviderForm {
	display: flex;
	flex-direction: column;
}

.Name-Provider {
	display: block;
	margin-top: 4px;
	margin-bottom: 4px;
	padding-left: 10px;
	height: 25px;
	width: 200px;
	border-radius: 10px;
	background: white;
	border-color: rgb(183, 185, 178);
	cursor: pointer;
}

.Lastname-Provider {
	display: block;
	margin-top: 4px;
	margin-bottom: 4px;
	padding-left: 10px;
	height: 25px;
	width: 200px;
	border-radius: 10px;
	background: white;
	border-color: rgb(183, 185, 178);
	cursor: pointer;
}

.Empresa-Provider {
	display: block;
	margin-top: 4px;
	margin-bottom: 4px;
	padding-left: 10px;
	height: 25px;
	width: 200px;
	border-radius: 10px;
	background: white;
	border-color: rgb(183, 185, 178);
	cursor: pointer;
}

.Nit-Provider {
	display: block;
	margin-top: 4px;
	margin-bottom: 4px;
	padding-left: 10px;
	height: 25px;
	width: 200px;
	border-radius: 10px;
	background: white;
	border-color: rgb(183, 185, 178);
	cursor: pointer;
}

.Correo {
	display: block;
	margin-top: 4px;
	margin-bottom: 4px;
	padding-left: 10px;
	height: 25px;
	width: 200px;
	border-radius: 10px;
	background: white;
	border-color: rgb(183, 185, 178);
	cursor: pointer;
}

.Password-Provider {
	display: block;
	margin-top: 4px;
	margin-bottom: 4px;
	padding-left: 10px;
	height: 25px;
	width: 200px;
	border-radius: 10px;
	background: white;
	border-color: rgb(183, 185, 178);
	cursor: pointer;
}

.SubmitProvider {
	display: block;
	margin-top: 5px;
	margin-left: 27px;
	height: 40px;
	width: 150px;
	border-radius: 20px;
	padding-top: 6px;
	font-weight: 600;
	background: #e0c408c9;
	border-color: black;
	color: white;
	cursor: pointer;
	text-decoration: none;
}

.SubmitProvider:hover {
	background: #e0c308;
}

@media screen and (min-width: 768px) {
	.FormContainer {
		width: 768px;
		height: 529px;
	}
	.PaginaNew-Provider {
		width: 290px;
		height: 250px;
	}
}

@media screen and (min-width: 992px) {
	.FormContainer {
		width: 1440px;
		height: 620px;
	}

	.NavBar-Logo-ProvidersCreate {
		width: 400px;
		margin: 15px;
	}

	.PaginaNew-Provider {
		width: 350px;
		height: 300px;
	}
	.Name-Provider {
		grid-gap: 30px;
		font-weight: 500;
	}

	.Lastname-Provider {
		font-weight: 500;
	}
	.Empresa-Provider {
		font-weight: 500;
	}

	.Nit-Provider {
		font-weight: 500;
	}

	.Correo {
		font-weight: 500;
	}

	.Password-Provider {
		font-weight: 500;
	}

	.SubmitProvider {
		font-weight: 500;
	}
}

@media screen and (min-width: 1200px) {
	.FormContainer {
		width: 1440px;
		height: 620px;
	}

	.NavBar-Logo-ProvidersCreate {
		width: 400px;
		margin: 15px;
	}

	.PaginaNew-Provider {
		width: 350px;
		height: 300px;
	}
	.Name-Provider {
		grid-gap: 30px;
		font-weight: 500;
	}

	.Lastname-Provider {
		font-weight: 500;
	}

	.Empresa-Provider {
		font-weight: 500;
	}

	.Nit-Provider {
		font-weight: 500;
	}

	.Correo {
		font-weight: 500;
	}

	.Password-Provider {
		font-weight: 500;
	}

	.SubmitProvider {
		font-weight: 500;
	}
}
