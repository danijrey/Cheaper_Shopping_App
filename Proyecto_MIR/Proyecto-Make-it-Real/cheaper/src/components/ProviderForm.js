import React from "react";
import logo from "../assets/logo.png";
import "./ProviderForm.css";

function ProviderForm({
  name,
  lastname,
  company,
  nit,
  providerEmail,
  password,
  handleChange,
  handleSubmit,
}) {
  return (
    <div className="App">
      <div className="FormContainer">
        <img
          src={logo}
          className="NavBar-Logo-ProvidersCreate"
          alt="Logo"
        ></img>
        <div className="PaginaNew-Provider">
          <h1 className="TitleProvider">Ingresa tus datos</h1>
          <div className="InputsNew-Provider">
            <form
              className="ProviderForm"
              onSubmit={handleSubmit}
              data-testid="ProviderForm"
            >
              <input
                className="Name-Provider"
                type="text"
                name="name"
                id="name"
                placeholder="Nombre"
                onChange={handleChange}
                value={name}
              />
              <input
                className="Lastname-Provider"
                type="text"
                name="lastname"
                id="lastname"
                placeholder="Apellidos"
                onChange={handleChange}
                value={lastname}
              />
              <input
                className="Empresa-Provider"
                type="text"
                name="company"
                id="company"
                placeholder="Empresa"
                onChange={handleChange}
                value={company}
              />
              <input
                className="Nit-Provider"
                type="number"
                name="nit"
                id="nit"
                placeholder="Nit"
                onChange={handleChange}
                value={nit}
              />
              <input
                className="Correo"
                type="email"
                name="providerEmail"
                id="providerEmail"
                placeholder="Correo"
                onChange={handleChange}
                value={providerEmail}
              />

              <input
                className="Password-Provider"
                type="password"
                name="password"
                id="password"
                placeholder="ContraseÃ±a"
                onChange={handleChange}
                value={password}
              />
              <button className="SubmitProvider" type="submit">
                Enviar
              </button>
            </form>
          </div>
        </div>
      </div>
    </div>
  );
}

export default ProviderForm;
