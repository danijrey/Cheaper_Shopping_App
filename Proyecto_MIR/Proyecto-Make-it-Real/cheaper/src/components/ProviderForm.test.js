import React from 'react';
import {render} from '@testing-library/react';
import ProviderForm from './ProviderForm';


describe ('ProviderForm',() =>{
    it ('should render correctly', ()=>{
    const {container} = render (<ProviderForm/>); 
    expect (container.firstChild).toMatchSnapshot();
    });
});