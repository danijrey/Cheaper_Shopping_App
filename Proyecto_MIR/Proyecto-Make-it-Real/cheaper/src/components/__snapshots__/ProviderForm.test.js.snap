// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProviderForm should render correctly 1`] = `
<div
  class="App"
>
  <div
    class="FormContainer"
  >
    <img
      alt="Logo"
      class="NavBar-Logo-ProvidersCreate"
      src="logo.png"
    />
    <div
      class="PaginaNew-Provider"
    >
      <h1
        class="TitleProvider"
      >
        Ingresa tus datos
      </h1>
      <div
        class="InputsNew-Provider"
      >
        <form
          class="ProviderForm"
          data-testid="ProviderForm"
        >
          <input
            class="Name-Provider"
            id="name"
            name="name"
            placeholder="Nombre"
            type="text"
            value=""
          />
          <input
            class="Lastname-Provider"
            id="lastname"
            name="lastname"
            placeholder="Apellidos"
            type="text"
            value=""
          />
          <input
            class="Empresa-Provider"
            id="company"
            name="company"
            placeholder="Empresa"
            type="text"
            value=""
          />
          <input
            class="Nit-Provider"
            id="nit"
            name="nit"
            placeholder="Nit"
            type="number"
            value=""
          />
          <input
            class="Correo"
            id="providerEmail"
            name="providerEmail"
            placeholder="Correo"
            type="email"
            value=""
          />
          <input
            class="Password-Provider"
            id="password"
            name="password"
            placeholder="ContraseÃ±a"
            type="password"
            value=""
          />
          <button
            class="SubmitProvider"
            type="submit"
          >
            Enviar
          </button>
        </form>
      </div>
    </div>
  </div>
</div>
`;
